From 8a6626d60d70edf779b22bebe35a00dffe63e511 Mon Sep 17 00:00:00 2001
From: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Date: Tue, 23 Jan 2024 15:20:29 +0900
Subject: [PATCH] hwdb: ieee1394-unit-function: adjustment of entries with
 device attributes available in Linux v6.8

The series of changes[1] has been merged to Linux v6.8-rc1[2], which
alters the parser of the content of configuration ROM. As a result, some
device attributes for model information in the legacy layout of
configuration ROM are finally available for node and unit devices. The
change enables to distinguish the devices by model information, and what
is required in systemd PR 30205[3]. It can improve the hwdb for IEEE 1394
functions in the point addressed at issue 25029[4].

This commit fulfills some entries for the hwdb, which corresponds to some
devices with the legacy layout of configuration ROM. They provides both
vendor and model names to udev applications.

[1] https://lore.kernel.org/lkml/20240110111331.GA138108@workstation.local/
[2] https://lore.kernel.org/lkml/CAHk-=wiB4iHTtfZKiy5pC24uOjun4fbj4kSX0=ZnGsOXadMf6g@mail.gmail.com/
[3] https://github.com/systemd/systemd/pull/30205
[4] https://github.com/systemd/systemd/issues/25029

(cherry picked from commit bd922111fd752fe39f1bb31a1ff7658d1a5db3ca)

Related: RHEL-5950
---
 hwdb.d/80-ieee1394-unit-function.hwdb | 32 ++++++++++++++++++++-------
 1 file changed, 24 insertions(+), 8 deletions(-)

diff --git a/hwdb.d/80-ieee1394-unit-function.hwdb b/hwdb.d/80-ieee1394-unit-function.hwdb
index 75129823bb..ce29c5b217 100644
--- a/hwdb.d/80-ieee1394-unit-function.hwdb
+++ b/hwdb.d/80-ieee1394-unit-function.hwdb
@@ -1342,18 +1342,34 @@ ieee1394:ven00001287mo00000036sp0000A02Dver00010001
 # 1.0 (December 12, 2000, 1394 Trading Association, TA Document 1999027)'.
 #
 
-ieee1394:node:ven0x080046units0x00a02d:0x010001
-ieee1394:ven00080046mo00000000sp0000A02Dver00010001
+# In Linux kernel v6.7 or before. The model field is not filled and left as zero.
+ieee1394:node:ven0x080046units0x00a02d:0x01000[01]
+ieee1394:ven00080046mo00000000sp0000A02Dver0001000[01]
  ID_VENDOR_FROM_DATABASE=Sony Corporation
- ID_MODEL_FROM_DATABASE=DCR-TRV310
  IEEE1394_UNIT_FUNCTION_VIDEO=1
 
-#
-# Sony Media converter without AV/C command set.
-#
+ieee1394:node:ven0x080046mo0xfa0000units0x00a02d:0x010000
+ieee1394:ven00080046mo00FA0000sp0000A02Dver00010000
+ ID_VENDOR_FROM_DATABASE=Sony
+ ID_MODEL_FROM_DATABASE=DVMC-DA1
+ IEEE1394_UNIT_FUNCTION_VIDEO=1
+
+ieee1394:node:ven0x080046mo0x380027units0x00a02d:0x010001
+ieee1394:ven00080046mo00380027sp0000A02Dver00010001
+ ID_VENDOR_FROM_DATABASE=Sony
+ ID_MODEL_FROM_DATABASE=DCR-TRV120
+ IEEE1394_UNIT_FUNCTION_VIDEO=1
+
+ieee1394:node:ven0x080046mo0x38000funits0x00a02d:0x010001
+ieee1394:ven00080046mo0038000Fsp0000A02Dver00010001
+ ID_VENDOR_FROM_DATABASE=Sony
+ ID_MODEL_FROM_DATABASE=DCR-TRV310
+ IEEE1394_UNIT_FUNCTION_VIDEO=1
 
-ieee1394:node:ven0x080046units0x00a02d:0x010000
-ieee1394:ven00080046*sp0000A02Dver00010000
+ieee1394:node:ven0x008045mo0x80a020units0x00a02d:0x010001
+ieee1394:ven00008045mo0080A020sp0000A02Dver00010001
+ ID_VENDOR_FROM_DATABASE=Panasonic
+ ID_MODEL_FROM_DATABASE=AG-DV1
  IEEE1394_UNIT_FUNCTION_VIDEO=1
 
 #
